### ISSUE TITLE
Compilation times out from time to time (huge application)

### ISSUE THREAD

**Issue by PragTob** [elixir-lang/elixir#12506]
### Elixir and Erlang/OTP versions

```
elixir -v
Erlang/OTP 25 [erts-13.1.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Elixir 1.14.3 (compiled with Erlang/OTP 25)
```

### Operating system

macOS 13.1

### Current behavior

Compilation will sometimes time out, seemingly around the end of compilation. This is on a very big phoenix code base, with one of our weaker laptops (intel MBP 13 inch, 4 cores, 2 Ghz or so)

It _may_ be related to 845c20a33524f1946a2660e5c7d4c8904cc79d30 but I couldn't check it out as on `main` one of our dependencies currently doesn't compile and I didn't have the time to start the yak 😓 

this is the stacktrace when it happens:

```
** (exit) exited in: :gen_server.call(#PID<0.741.0>, :start)
    ** (EXIT) time out
    (stdlib 4.2) gen_server.erl:370: :gen_server.call/2
    (elixir 1.14.3) lib/kernel/parallel_compiler.ex:262: Kernel.ParallelCompiler.verify_modules/4
    (elixir 1.14.3) lib/kernel/parallel_compiler.ex:204: Kernel.ParallelCompiler.spawn_workers/5
    (elixir 1.14.3) lib/kernel/parallel_compiler.ex:177: Kernel.ParallelCompiler.spawn_workers/3
    (mix 1.14.3) lib/mix/compilers/elixir.ex:158: Mix.Compilers.Elixir.compile/7
    (mix 1.14.3) lib/mix/state.ex:15: Mix.State.lock/2
    (mix 1.14.3) lib/mix/task.ex:421: anonymous fn/3 in Mix.Task.run_task/4
```

Compiling again usually solves the issue. Sometimes the error occurs again and another compile is necessary.

### Expected behavior

No errors :)

Thanks again for all your amazing work as usual! 💚

**Comment by josevalim** [comment]
This has been fixed on v1.14 and main branches. We should have a new release sooner than later (just ironing out the remaining issues with Erlang/OTP 26). :)

**Comment by PragTob** [comment]
Awesome, thank you! 💚 

![IMG_20180818_153035](https://user-images.githubusercontent.com/606517/229201744-40dd5daf-6de1-4cfe-9ce4-5c6b4fbcc399.jpg)
